generators/directive/index.js

```diff
@@ -33,115 +38,89 @@ module.exports = class extends Generator {
-        properties
+        this.templatePath('directive.scss'),
+        this.destinationPath(`src/directives/${opts.urlSafe}/${opts.urlSafe}.scss`),
+        opts
       );
-
-    }else{
-
+    } else {
       this.fs.copyTpl(
-        getTemplate('directiveAttribute/directive.js'),
-        this.destinationPath(`src/directives/${properties.prefix}-${properties.urlSafe}/${properties.prefix}-${properties.urlSafe}.directive.js`),
-        properties
+        this.templatePath('attr-directive.js'),
+        this.destinationPath(`src/directives/${opts.urlSafe}/${opts.urlSafe}.directive.js`),
+        opts
       );
-
     }
 
-    var yet = this;
-
-    fs.readFile(yet.destinationPath('src/js/app.js'), 'utf-8', function(err, data){
-      if (err) yet.log(err);
+    // update the app file with the new module dependency
+    p1 = fsp
+      .readFile(this.destinationPath(appFile), 'utf8')
+      .then((data) => {
+        let newFile = data.replace(/(\/\/!!D!!\/\/)/, `'${opts.module}',\n\t\t\t//!!D!!//`);
 
-      var newValue = data.replace(/(\/\/!!D!!\/\/)/, '\''+properties.prefix + properties.camelCase+'Directive\', \n\t\t\t//!!D!!//');
-
-      fs.writeFile(yet.destinationPath('src/js/app.js'), newValue, 'utf-8', function (err) {
-        if (err) yet.log(err);
-        yet.log('Updated module dependencies')
-      });
-    });
-
-    // matches
-    //
-    // module('app', [
-    //  'test',
-    //  'tester2'
-    // ]);
-    //
-    // \.module\('app', \[(\n\s.+)+\]\)
-
-
-        // matches
-        //
-    // [
-    //  'test',
-    //  'tester2'
-    // ]
-
-    // \[(\n\s.+)+\]
-
-    fs.readFile(yet.destinationPath('src/scss/styling/_components.scss'), 'utf-8', function(err, data){
-      if (err) yet.log(err);
-
-      var newValue = data.replace(/(\/\/!!D!!\/\/)/, '@import "../../directives/'+properties.prefix + '-' + properties.urlSafe + '/'+properties.prefix + '-' + properties.urlSafe + '"; \n//!!D!!//');
-
-      fs.writeFile(yet.destinationPath('src/scss/styling/_components.scss'), newValue, 'utf-8', function (err) {
-        if (err) yet.log(err);
-        yet.log('Injected styling into components file');
+        return fsp.writeFile(this.destinationPath(appFile), newFile);
+      })
+      .then(() => {
+         this.log(`   ${chalk.green('update')} ${appFile}`);
       });
-    });
 
+    if (opts.element) {
+      // update the components sass file with the new component
+      p2 = fsp
+        .readFile(this.destinationPath(compFile), 'utf8')
+        .then((data) => {
+          let newFile = data.replace(/(\/\/!!D!!\/\/)/, `@import "../../directives/${opts.urlSafe}/${opts.urlSafe}";\n//!!D!!//`);
+
+          return fsp.writeFile(this.destinationPath(compFile), newFile);
+        })
+        .then(() => {
+          this.log(`   ${chalk.green('update')} ${compFile}`);
+        });
+      }
```

Comment:
as with the module dependency updater, using `fs-promise` i think cleans up the code and simplifies it. errors are automatically dropped to the console if they occur.

changing the log output more closely matches the generator's other output, slightly improving the UX.